contracts/protocol/pool/Pool.sol

```diff
@@ -156,7 +156,8 @@ contract Pool is VersionedInitializable, PoolStorage, IPool {
         asset: asset,
         amount: amount,
         onBehalfOf: onBehalfOf,
-        referralCode: referralCode
+        referralCode: referralCode,
+        addressesProvider: address(ADDRESSES_PROVIDER)
```

Comment:
I hate the need of passing the addressProvider address tbh, but didn't find a workaround

Reply:
I think we should do it, just the same way as we do for flashLoans (we are passing if its an authorized flash borrower)

Reply:
We thought about doing this, but in majority of cases the gas would be wasted as only in the case of a ceiled asset that is not already supplied the role needs to be fetched.